geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/SchemaHelpers.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2014 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.core.index
+
+import com.vividsolutions.jts.geom.{Geometry, GeometryCollection, Polygon}
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.{DateTime, DateTimeZone, Interval}
+import org.locationtech.geomesa.utils.text.WKTUtils
+
+import scala.annotation.tailrec
+import scala.util.parsing.combinator.RegexParsers
+
+trait SchemaHelpers extends RegexParsers {
+
+  val minDateTime = new DateTime(0, 1, 1, 0, 0, 0, DateTimeZone.forID("UTC"))
+  val maxDateTime = new DateTime(9999, 12, 31, 23, 59, 59, DateTimeZone.forID("UTC"))
+  val everywhen = new Interval(minDateTime, maxDateTime)
+  val everywhere = WKTUtils.read("POLYGON((-180 -90, 0 -90, 180 -90, 180 90, 0 90, -180 90, -180 -90))").asInstanceOf[Polygon]
+
+  val DEFAULT_TIME = new DateTime(0, DateTimeZone.forID("UTC"))
+
+  val CODE_START = "%"
+  val CODE_END = "#"
+  val GEO_HASH_CODE = "gh"
+  val DATE_CODE = "d"
+  val CONSTANT_CODE = "cstr"
+  val RANDOM_CODE = "r"
+  val SEPARATOR_CODE = "s"
+  val ID_CODE = "id"
+  val PART_DELIMITER = "::"
+
+  def somewhen(interval: Interval): Option[Interval] =
+    interval match {
+      case null                => None
+      case i if i == everywhen => None
+      case _                   => Some(interval)
+    }
+
+  def innerSomewhere(geom: Geometry): Option[Geometry] =
+    geom match {
+      case null                 => None
+      case p if p == everywhere => None
+      case g: Geometry          => Some(g)
+      case _                    => None
+    }
+
+  // This function helps catch nulls and 'entire world' polygons.
+  def somewhere(geom: Geometry): Option[Geometry] =
```

Comment:
maybe convert to PartialFunction[Geometry, Option[Geometry]] and get rid of the match statement